framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -593,195 +484,306 @@ public void standardizeVariable(AnnotatedTypeMirror type, Element elt) {
-        standardizeAtm(context, localVarPath, type, /*removeErroneousExpressions=*/ false);
+    protected void convertAnnotatedTypeMirror(
+            StringToJavaExpression stringToJavaExpr, AnnotatedTypeMirror type) {
+        this.annotatedTypeReplacer.visit(type, anno -> map(stringToJavaExpr, anno));
     }
 
     /**
-     * Parse the dependent types in {@code type} as if they were written at {@code localVarPath} and
-     * viewpoint-adapt to the given context.
+     * If {@code anno} is not a dependent type annotation, {@code null} is returned. Otherwise, this
+     * method applies {@code stringToJavaExpr} to all expressions strings in {@code anno} and then
+     * calls {@link #buildAnnotation(AnnotationMirror, Map)} to build a new annotation with the
+     * {@code JavaExpression}s converted back to strings.
+     *
+     * <p>If {@code stringToJavaExpr} returns {@code null}, then that expression is removed from the
+     * returned annotation.
      *
-     * @param context JavaExpressionParseContext
-     * @param localVarPath if non-null, the expression is parsed as if it were written at this
-     *     location
-     * @param type the type to "standardize"
-     * @param removeErroneousExpressions if true, remove erroneous expressions rather than
-     *     converting them into an explanation of why they are illegal
+     * <p>Instead of overriding this method subclasses can override the following methods to change
+     * the behavior of this class:
+     *
+     * <ul>
+     *   <li>{@link #shouldParseExpression(String)}: to control which expressions are skipped. If
+     *       this method returns false, then the expression string is not parsed and is included in
+     *       the new annotation unchanged.
+     *   <li>{@link #transform(JavaExpression)}: make changes to the JavaExpression produced by
+     *       {@code stringToJavaExpr}.
+     *   <li>{@link #buildAnnotation(AnnotationMirror, Map)}: to change the annotation returned by
+     *       this method.
+     * </ul>
+     *
+     * @param stringToJavaExpr function that converts strings to {@code JavaExpression}s
+     * @param anno annotation mirror
+     * @return an annotation created by applying {@code stringToJavaExpr} to all expressions strings
+     *     in {@code anno}
      */
-    private void standardizeAtm(
-            JavaExpressionContext context,
-            TreePath localVarPath,
-            AnnotatedTypeMirror type,
-            boolean removeErroneousExpressions) {
-        this.standardizeTypeAnnotator.init(context, localVarPath, removeErroneousExpressions);
-        this.standardizeTypeAnnotator.visit(type);
+    protected @Nullable AnnotationMirror map(
+            StringToJavaExpression stringToJavaExpr, AnnotationMirror anno) {
+        if (!isExpressionAnno(anno)) {
+            return null;
+        }
+
+        Map<String, List<JavaExpression>> map = new HashMap<>();
+        for (String value : getListOfExpressionElements(anno)) {
```

Comment:
Here is another place that I think `ExecutableElement` would be more appropriate (and more efficient) than `String`.